Yoast/javascript
# Remove all occurrences of createComponentWithIntl

## Summary

## Relevant technical choices:

* [algolia-search-box] Removes intl from `SearchResultDetail.js`.
* [helpers] Removes `packages/helpers/src/tests/createComponentWithIntl.js` and all occurrences of createComponentWithIntl in the tests.
* [`apps/components`, `apps/content-analysis`, algolia-search-box, helpers] Remove all remnants of react-intl.
* [analysis-report, components, helpers, search-metadata-previews, yoast-components] Removes all occurrences of createComponentWithIntl in the tests.

## Test instructions

This PR can be tested by following these steps:

* Run the tests and see them pass.
* Link it to the plugin, and make sure all strings in the SearchResultDetail component in the Algolia search in the help center are still present and translated: `Search result`, `View in KB`, `Open the knowledge base article in a new window or read it in the iframe below` (= aria-label), `Go back`, `Go back to the search results` (= aria-label), `Knowledge base article`.
* Build the two example apps and see no intl-related errors.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * The SearchResultDetail component in the Algolia search in the help center.

## UI changes

## Quality assurance


Fixes #271
